src/gc/gcpriv.h

```diff
@@ -781,6 +779,9 @@ class generation
     int             gen_num;
 
 #ifdef FREE_USAGE_STATS
+    #pragma message ("FREE_USAGE_STATS changes the layout of the 'generation' class, which is known to the DAC. " \
```

Comment:
Yeah, its still doable, just extra work to do the pointer arithmetic yourself. Other options would be: 
a) including the additional fields on generation always but not using them in some builds
b) using an indirection so that you add one pointer to generation always, but then use of that pointer is optional depending on build type. Then add any number of optional fields in the data structure the pointer points to and it won't affect the layout that DAC deals with.

Any of the options are fine from my end, just different tradeoffs of runtime memory overhead in the GC vs implementation complexity.